lib/rubygems/commands/fetch_command.rb

```diff
@@ -60,7 +60,7 @@ def execute
         specs_and_sources = filtered unless filtered.empty?
       end
 
-      spec, source = specs_and_sources.max_by {|s,| s.version }
+      spec, source = specs_and_sources.sort_by {|s,| s }.last
```

Comment:
I think `max_by` should still work:
```suggestion
      spec, source = specs_and_sources.max_by {|s,| s }
```

Reply:
Totally, not sure what I was thinking!